[CI] Cleanup .github/ folder

NlightNFotis opened the pull request:
This PR is removing some files that are no longer needed from the `.github/` folder,
and it's rewriting the `README.md` file in that folder to more accurately represent
the current state of affairs.

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

Commit:
Remove extraneous files and folders from .github

We have changed our implementation of package building
in CI, and some of the original implementation files
are no longer relevant.

Commit:
Update github actions readme file.

The original readme.md file is no longer corresponding
to the current implementation of CI we have, so this is
bringing it more in line with the current state.

tautschnig commented on .github/workflows/README.md:
```diff
@@ -1,56 +1,22 @@
-# CBMC packages
```
Should this information, rather than being removed, be moved to some other place? Perhaps even be merged into the yaml file that takes care of building installation packages?

NlightNFotis replied:
I do not think so. Most of that information is no longer relevant, and the relevant pieces of information (installation instructions) are present on both our release pages and the top-level `README.md` file.

thomasspriggs commented on .github/workflows/README.md:
```diff
@@ -1,56 +1,22 @@
+* `pull-request-check-clang-format.sh` -> Script that's called per Pull Request
+  to execute `clang-format` over changes.
+* `pull-request-checks.yaml` -> Configuration file for the Github Actions CI jobs
+  for the various platforms.
```
:pick: Might be nice to have a list of which platforms/build systems we are currently testing on pull requests.

NlightNFotis committed:
Rename some actions and add information on the CI platforms.

Add some information to the readme with regard to the main
platforms we are building/testing CBMC on Github Actions.
